src/core/mac/mac.cpp

```diff
@@ -269,7 +269,11 @@ ThreadError Mac::SetNetworkName(const char *aNetworkName)
     VerifyOrExit(strlen(aNetworkName) <= OT_NETWORK_NAME_MAX_SIZE, error = kThreadError_InvalidArgs);
 
     memset(&mNetworkName, 0, sizeof(mNetworkName));
+#ifdef _WIN32
+    strcpy_s(mNetworkName.m8, sizeof(mNetworkName), aNetworkName);
+#else
```

Comment:
Agreed about `strncpy()`. Indeed, the given usage is actually incorrect! Not sure why I didn't notice earlier.

I think the appropriate replacement function would be [`strlcpy`](https://www.freebsd.org/cgi/man.cgi?query=strlcpy&sektion=3), which works as everyone expects it should. It is available on BSD platforms, and its behavior is easily replicated in a replacement implementation.

That will fix the bug on all platforms.